### DISCUSSION TITLE
Where do we go for help?

### DISCUSSION THREAD

**Discussion by ecbaldwin** [csingley/ofxtools#160] [Ideas]
I just read [the announcement about OFX Home sunsetting](http://www.ofxhome.com/ofxforum/viewtopic.php?id=49843). I'm just entering into this struggle of trying to make automatic downloads of OFX data into my finance software work and finding it overwhelming.

This is a bigger issue than [simply deprecating `--ofxhome` from ofxget](https://github.com/csingley/ofxtools/issues/150) which is why I chose to file a new issue. I'm trying to make this work for institutions like Citi, Chase, and First Tech FCU but it doesn't seem possible due to what appears to be institution-specific issues. The help from the forum is read-only and quickly growing out of date and stale. For example, I went through all of the help from the [Chase forum](http://www.ofxhome.com/ofxforum/viewtopic.php?id=47456) and continue to run into issues. I ran into what seems to be new problems that are not mentioned and it isn't possible to post new content. So, where do we go for help from the community?

At the least, the [documentation for ofxget](https://ofxtools.readthedocs.io/en/latest/client.html) should reflect the fact that OFX Home is unmaintained and quickly becoming defunct. It still reads like it is the place to go. Other than that, I think this community needs a new lifeline to connect us together and get help and a new place to maintain the data that was provided by it. Could it be as simple as a GitHub repo containing the data?

**Comment by csingley** [comment]
First, would you please be so kind as to not open up tickets along these lines in the bug tracker?  I'd feel rude just closing your ticket with "NOTABUG", which is the appropriate response.

I feel sympathy for your plight, which you have expressed clearly.  I hope you'll allow me to express mine.

I am not the slightest bit interested in downloading OFX data, much less importing that data into other pieces of financial software.  These are just not needs that I have.  What I am interested in is _parsing_ OFX data, and _developing_ pieces of financial software.  If your interests are aligned with those, we're going to get along with other just great - although, again, the bug tracker is not the right place to kick off that relationship.

I published the `ofxget` client pretty much just because, well, I'm a nice guy, and it wasn't too much work for me to do - another 5%, perhaps, on top of the vastly larger effort of developing a parser and object model for most of the OFX spec, which is not a small document.

It hadn't really occurred to me that there's almost nobody who shares my interest in actually developing software that uses OFX data, but there's a pretty hefty population of people who want to use an OFX client to download data so it can be parsed by Gnucash or Moneydance or what have you... and a high percentage of *those* people either can't or don't want to modify the `ofxtools` source to suit their needs.

That use case is kind of a nightmare to support and keep running, as you're finding out.  Nobody really wants to do it - I certainly don't.  It's kind of a lost cause anyway, since the OFX consortium has thrown in their lot with a bunch of wankers who won't let you download your data at all.  It's just  a matter of time before OFX download clients are obsolete.

If we accept the premise that what you're looking for is in fact a good use of resources - which it may be, or may not - I think you'll find that what's lacking here is manpower & brainpower, not really leadership.

Again, I think I'm a nice guy, but I believe I can make a strong case that I have in fact already done my part, for more than a decade now.  I'm pretty busy these days.

If it's getting under your skin that the ofxtools docs are getting stale... by all means, fix them up... nothing I like better than a pull request... or else do please be patient until I get around to doing it myself.

If you'd like to keep ofxhome going, or get the data and do something with it... talk to him about it, he's also a nice guy, and I'm sure would be willing to work with you.

Sorry to unburden myself... but you know, there are other actual bugs sitting on this tracker... even if they are bugs filed against the OFX client I don't use.   Cause, you know, @aclindsa does indeed make contributions to ofxtools, and that tends to get my attention.

**Reply by ecbaldwin** [reply]
I guess I hit a nerve.

> First, would you please be so kind as to not open up tickets along these lines in the bug tracker?

Maybe it wasn't really concisely written but mostly I filed an issue because I thought the documentation for this project, which doesn't seem to be in the repo, could use an update to reflect the current situation. I couldn't find an issue tracker specifically for that. I apologize for breaking protocol but sometimes when there isn't one, you just take a stab, you know?

> I am not the slightest bit interested in downloading OFX data, much less importing that data into other pieces of financial software. These are just not needs that I have. What I am interested in is parsing OFX data, and developing pieces of financial software

I'm sorry but this doesn't make sense. Without data and financial software, your little project exists in a vacuum and has no reason to. I understand that you're frustrated and you're doing your part. But, this doesn't give much hope. I see this attitude everywhere I've looked for help on this subject.

I'm just trying to reach out to see what kind of community there is but it sounds like it has been frustrated to the point of not wanting to exist. Oh well.

**Reply by aclindsa** [reply]
> I'm sorry but this doesn't make sense. Without data and financial software, your little project exists in a vacuum and has no reason to.

I don't believe he is saying he doesn't care about the data. My understanding is that he 1) obtains the OFX content in ways outside of `ofxget` and 2) processes the parsed data on his own, without a need to import it into third-party software. While I and many others tend to want what you want (downloading OFX content from financial institutions for importing into third-party financial management software), it's not the only conceivable use for OFX-parsing software.

> it has been frustrated to the point of not wanting to exist

This may not be far from the truth. Even leaving aside the fact that @csingley is maintaining parts of his project he doesn't personally care about... It is a real slog to maintain automatic imports of financial data using OFX from all my financial institutions. Several of them have given up on "Direct Connect" altogether, others are following soon, those that remain have so many "quirks" in their implementations that it's a real pain, and new "quirks" are frequently introduced to keep you on your toes. There are very, very few people still willing to fight this losing battle.

I very much share your desire to fight for the ability to openly and directly access our own financial data, and I even like the idea of some sort of community. Unfortunately, the intersection of 1) people who care about their finances enough to want this data 2) don't want to use an existing solution provided by someone else, 3) have the ability, 4), time 5) and desire to contribute to an open-source solution is vanishingly small.

> Could it be as simple as a GitHub repo containing the data?

Are you interested in maintaining this? If so, go for it! I'll second the suggestion to reach out to ofxhome (though it's not exactly clear to be how to do so). I don't know how long this will be useful as FI's increasingly transition to FDX and effectively lock us out, but I'm not quite done fighting yet and would occasionally contribute when FI's break things in new and exciting ways that we have to fix.

**Comment by csingley** [comment]
Friend, perhaps you did catch me on a bad day... but I'm none too pleased about the apparent trend for all comers to use my bug tracker as some sort of support forum, totally divorced from the goal of improving the code quality of ofxtools.

Try to look at this from my point of view.  If you had a product, and you built a bonus feature on it, and suddenly that (utterly noncore) 5% of your code started producing 95% of the headaches, you'd think long & hard about cutting that dog loose, wouldn't you?  Now open source has been good to me, so I'm not really that transactional about it... but good god in heaven above, I certainly am not looking to MASSIVELY INCREASE that particular time suck.  Full power to shields!!

Apparently I don't share the same POV as nearly all my other users - but look, I was trained as a financial professional, and you had better believe I am maintaining backup documentation for every transaction I book and every position that I take.  That means I have to shoulder the burden of a regular cycle of logging into every FI website and downloading PDF statements... unless you're super into hard-copy filing systems, the PDFs are pretty much a hard requirement, if you have any game plan beyond throwing yourself upon the mercy of an auditor when the feces impacts the ventilator.  That being the case, it's really not much extra hassle to click the next tab over on the FI website and select "Download Microsoft Money" or what have you.  Then you also have an archive of the primary data, uncontaminated by the flaws introduced by you or your financial software when you imported it.

So sure, automatic downloads & imports are great and all, but it's like a 2 on a scale from 1-10.  I don't get what all the fuss and the gnashing of teeth is about.  The writing is on the wall, the OFX servers are going to be gradually turned off in the near future... but I guess we'll still be able to click "Download Quicken format" for quite some time (just as we have with "Microsoft money") since the FIs don't need to pay Quicken or FiServ fat fees for that.

What I have really built here... an 11 on the 10-point scale... isn't the tinkertoy web client, it's quite a full-fledged OFX parser (a lot better than libofx) and OFX generator.  And it's in Python, which makes it pretty painless for amateur programmers (like me!) to rapidly develop custom financial software.  I mostly use ofxtools to do taxes, portfolio management, etc.  Chores where the commercial software is expensive & sucky, and the free software doesn't really cut the mustard.  I've got some hope that Martin from beancount will eventually get there, but it's probably going to take him years (and significantly larger quantities of money to manage) before he comes around to the necessity of building something great.

I really knew almost nothing about computer programming when I started working on this - like I said, I'm a finance guy by training, not a software guy.  But I needed it, so I buckled down, figured it out, kept grinding, and built it.  A big part of the reason it is open source is to attract like-minded people to participate in the work... and that's been great!

If you're just now becoming aware of the parlous state of the world as regard to free, open standard access to your financial data.... I can only quote John McLane:  "Welcome to the party pal!"  If you want anything to happen, I can only advise you to grab a shovel & start digging.

But I was telling the truth when I said you have my sympathy.  If you're looking for a water cooler to hang around & complain about OFX ... even if you're just looking to "Exhort inspire moral suasion, let's you & him do a lot of work"... this is indeed a place where that's welcome.  Just, y'know, keep it off the bug tracker, that's starting to piss me off.

Hope you manage to liberate your data!

**Reply by lnguyen639** [reply]
I admire you for saying "shit hits the fan" in the nicest way possible :D
